index.bs

```diff
@@ -3331,6 +3331,207 @@ be the same.
+
+    <pre highlight="webidl">
+        interface A {
+          long calculateWithType(long input);
+          DOMString calculateWithType(DOMString input);
+          CalculatableInterface calculateWithType(CalculatableInterface input);
+        };
+    </pre>
+
+    than using a [=union type=] with a [=typedef=] as
+
+    <pre highlight="webidl">
+        typedef (long or DOMString or CalculatableInterface) Calculatable;
+        interface A {
+          Calculatable calculateWithType(Calculatable input);
+        };
+    </pre>
+
+    which does not convey the fact that the return value is always of the same type as |input|.
+
+    The problem is exacerbated when one of the overloads has a return type of {{void}}, since
+    [=union types=] cannot even contain {{void}} as a [=member type=]. In that case, a return type
+    of {{any}} must be used with appropriate prose defining the return type, further decreasing
+    expressiveness.
+
+*   When the operation has significantly different semantics for different argument types or
+    lengths, [=overloaded|overloading=] is preferred.
+
+    As an example, the {{CSS/supports()}} operation of the {{CSS}} interface is defined as the
+    following IDL fragment [[CSS3-CONDITIONAL]].
+
+    <pre highlight="webidl">
+        partial interface CSS {
+          static boolean supports(CSSOMString property, CSSOMString value);
+          static boolean supports(CSSOMString conditionText);
+        };
+    </pre>
+
+    Using [=optional arguments=] one can rewrite the IDL fragment as follows:
+
+    <pre highlight="webidl">
+        partial interface CSS {
+          static boolean supports(CSSOMString propertyOrConditionText, optional CSSOMString value);
+        };
+    </pre>
+
+    It can be observed that, even though the IDL is shorter, two distinctively different concepts
+    are conflated in the first argument. This makes the second version remarkably less readable than
+    the first.
+
+    Another consideration is that many specification preprocessors allow linking to two different
+    prose definitions in case of [=overloaded|overloading=] but not [=optional arguments=]. This
+    means that in the first case the specification author may write the prose definition of the
+    operations as:
+
+    <div algorithm="execute supports() with overloads">
+
+        The <code class="idl">supports(<var ignore>property</var>, <var ignore>value</var>)</code>
+        method, when called, must run these steps:
+
+        1.  …
+
+        ----
+
+        The <code class="idl">supports(<var ignore>conditionText</var>)</code> method, when called,
+        must run these steps:
+
+        1.  …
+
+    </div>
+
+    Yet using optional argument, the specification author must use more boilerplate-style text to
+    effectively replicate the overload logic. If the two overloads have little to no shared parts,
+    it is better to leave overload resolution to the IDL mechanism.
```

Comment:
Let's move the conclusion sentence after this next algorithm.

Should "overload logic" link to the algorithm?